apps/opik-backend/src/main/java/com/comet/opik/domain/SpanDAO.java

```diff
@@ -858,6 +980,17 @@ private Mono<List<Span>> enhanceWithFeedbackScores(List<Span> spans) {
                 .doFinally(signalType -> endSegment(segment));
     }
 
+    private double calculateCost(Span span) {
+        // Later we could just use span.model(), but now it's still located inside metadata
+        String model = StringUtils.isNotBlank(span.model())
+                ? span.model()
+                : Optional.ofNullable(span.metadata())
+                        .map(metadata -> metadata.get("model"))
+                        .map(JsonNode::asText).orElse("");
+
+        return ModelPrice.fromString(model).calculateCost(span.usage());
+    }
```

Comment:
This should be calculated at the Service layer

Reply:
I left it here, since we also calculate cost for updates, and it's done in DAO level, Update POJO doesn't even have cost field. So to be consistent for create/update it's better to keep it here. Also we already have similar functions in DAO, like `enhanceWithFeedbackScores`, and if we move this to Service layer, then `enhanceWithFeedbackScores` should be moved too. 
Such refactoring now for would take additional time, and I prefer to keep as is, looks more incapsulated.

Reply:
Agree with this, although we might compromise.